packages/web-components/@sfx/products/src/products.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  customElement,
+  html,
+  property,
+  TemplateResult,
+} from 'lit-element';
+import { ProductModel } from '@sfx/product';
+import { Base } from '@sfx/base';
+
+/** The name of the event that contains product data. */
+export const PRODUCTS_EVENT: string = 'sfx::provide_products'
+
+/**
+ * The `sfx-products` web component wraps and renders a number of
+ * `sfx-product` components. It wraps each `sfx-product` component in an
+ * additional wrapper for flexibility.
+ */
+@customElement('sfx-products')
+export default class Products extends Base {
+  @property({ type: Array }) products: ProductModel[] = [];
+
+  /**
+   * Binds relevant methods.
+   */
+  constructor() {
+    super();
+
+    this.setProductsFromEvent = this.setProductsFromEvent.bind(this);
+  }
+
+  /**
+   * Registers event listeners.
+   */
+  connectedCallback() {
+    super.connectedCallback();
+
+
+    if (!this.getAttribute('role')) {
+      this.setAttribute('role', 'list');
```

Comment:
What does the `role` actually do? Should `role` be a property?

Reply:
`role` is the [ARIA role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) for accessibility. Since this component is effectively a list and we aren't using a real list element like `ul`, setting `role` to `list` signals to the client that this element acts as a list.

The `HTMLElement` interface does not expose `role` as a property, so it must be set as an attribute.

Reply:
@DanielMoniz-GBI Question: Should we declare this attribute earlier in the constructor ? I see examples in lit-element where we declare properties/attributes as static getters and then set them in the constructor?  https://lit-element.polymer-project.org/guide/properties#observed-attributes Just wondering